src/storage/src/source/statistics.rs

```diff
@@ -15,10 +15,136 @@ use std::rc::Rc;
 use timely::progress::frontier::Antichain;
 use timely::progress::Timestamp;
 
+use mz_ore::metric;
+use mz_ore::metrics::{CounterVecExt, DeleteOnDropCounter, DeleteOnDropGauge, GaugeVecExt};
+use mz_ore::metrics::{IntCounterVec, MetricsRegistry, UIntGaugeVec};
 use mz_repr::GlobalId;
 use mz_storage_client::client::SourceStatisticsUpdate;
+use prometheus::core::AtomicU64;
 
-/// A helper struct designed to make it easy for operators to update metrics
+use crate::source::metrics::SourceBaseMetrics;
+
+#[derive(Clone, Debug)]
+pub(crate) struct SourceStatisticsMetricsDefinitions {
+    pub(crate) snapshot_committed: UIntGaugeVec,
+    pub(crate) messages_received: IntCounterVec,
+    pub(crate) updates_staged: IntCounterVec,
+    pub(crate) updates_committed: IntCounterVec,
+    pub(crate) bytes_received: IntCounterVec,
+}
+
+impl SourceStatisticsMetricsDefinitions {
+    pub(crate) fn register_with(registry: &MetricsRegistry) -> Self {
+        Self {
+            snapshot_committed: registry.register(metric!(
+                name: "mz_snapshot_committed",
+                help: "Whether or not the worker has committed the initial snapshot for a source.",
+                var_labels: ["source_id", "worker_id", "parent_source_id", "shard_id"],
```

Comment:
Why do we also add the `parent_source_id` in all these. You think it will just be more convenient like this?

Reply:
Yes! Otherwise, there is gonna be now way in grafana to say: "give me all the data for this pg source"

In general: more labels is fine, as long as they don't add more time-series. Adding this label won't ever cause a timeseries for say one sub-source persist_sink to turn into 2 time-series so we are fine!